knimin/handlers/ag_pulldown.py

```diff
@@ -67,8 +84,42 @@ def post(self):
         failtext = ("The following barcodes were not retrieved "
                     "for any survey:\n%s" % failed)
         meta_zip.append("failures.txt", failtext)
+
+        # check database about what surveys are available
+        available_agsurveys = {}
+        for (id, name, selected) in db.list_ag_surveys():
+            available_agsurveys[id] = name.replace(' ', '_')
+
+        results_as_pd = []
         for survey, meta in viewitems(metadata):
-            meta_zip.append('survey_%s_md.txt' % survey, meta)
+            # only create files for those surveys that have been selected by
+            # the user. Note that ids from the DB are negative, in metadata
+            # they are positive!
+            # Currently, I (Stefan Janssen) don't have test data for external
+            # surveys, thus I don't know their 'survey' value. I expect it to
+            # be the name of the external survey. In order to not block their
+            # pulldown I check that a skipped survey ID must be in the set of
+            # all available surveys.
+            if ((-1 * survey) in selected_ag_surveys) or \
```

Comment:
`abs_survey = abs(survey)` and use `abs_survey` instead of `-1 * survey` in order to avoid code duplication